.jbang/CheckoutPR.java

```diff
@@ -107,4 +146,55 @@ public static void main(String[] args) throws Exception {
 
         System.out.println("Checked out PR #" + pr.getNumber() + " (" + pr.getTitle() + ") to branch '" + localBranchName + "'.");
     }
+
+    private static void checkoutUpstreamMain() throws Exception {
+        File repoDir = new File(".");
+        Repository repository = new FileRepositoryBuilder()
+                .setGitDir(new File(repoDir, ".git"))
+                .readEnvironment()
+                .findGitDir()
+                .build();
+
+        try (Git git = new Git(repository)) {
+            final String upstreamName = "upstream";
+            final String jabrefRepoUrl = "https://github.com/JabRef/jabref.git";
+
+            // Check if a remote pointing to JabRef/jabref already exists
+            List<RemoteConfig> remotes = git.remoteList().call();
+            Optional<RemoteConfig> jabrefRemote = remotes.stream()
+                // We use "contains", because there could be SSH remote URLs
+                .filter(r -> r.getURIs().stream().anyMatch(uri -> uri.toString().contains("JabRef/jabref")))
+                .findFirst();
+
+            String remoteToUse;
+            if (jabrefRemote.isPresent()) {
+                remoteToUse = jabrefRemote.get().getName();
+                System.out.println("Using existing remote: " + remoteToUse);
+            } else {
+                System.out.println("Adding remote 'upstream' pointing to " + jabrefRepoUrl);
+                git.remoteAdd()
+                    .setName(upstreamName)
+                    .setUri(new URIish(jabrefRepoUrl))
+                    .call();
+                remoteToUse = upstreamName;
+            }
+
+            // If current branch is not "main", checkout "main"
+            String currentBranch = repository.getBranch();
+            if (!"main".equals(currentBranch)) {
+                System.out.println("Checking out 'main' branch");
+                git.checkout().setName("main").call();
+            }
+
+            // Fetch from the selected remote
+            System.out.println("Fetching from " + remoteToUse);
+            String[] jGitArgsFetch = {"fetch", remoteToUse};
+            org.eclipse.jgit.pgm.Main.main(jGitArgsFetch);
```

Comment:
Can't we use the JGit apis here as well?
```suggestion
            git.fetch().setRemote(remoteToUse).call();
```